pkg/skaffold/deploy/helm/args.go

```diff
@@ -201,3 +204,53 @@ func (h *Deployer) installArgs(r latest_v1.HelmRelease, builds []graph.Artifact,
 
 	return args, nil
 }
+
+// sanitizeFilePath is used to sanitize filepaths that are provided to the `setFiles` flag
+// helm `setFiles` doesn't work with the unescaped filepath separator (\) for Windows or if there are unescaped tabs and spaces in the directory names.
+// So we escape all odd count occurrences of `\` for Windows, and wrap the entire string in quotes if it has spaces.
+// This is very specific to the way helm handles its flags.
+// See https://github.com/helm/helm/blob/d55c53df4e394fb62b0514a09c57bce235dd7877/pkg/cli/values/options.
+// Otherwise the windows `syscall`package implements its own sanitizing for command args that's used by `exec.Cmd`.
+// See https://github.com/golang/go/blob/6951da56b0ae2cd4250fc1b0350d090aed633ac1/src/syscall/exec_windows.go#L27
+func sanitizeFilePath(s string) string {
+	needsQuotes := false
+	if len(s) == 0 {
+		return `""`
+	}
+
+	for i := 0; i < len(s); i++ {
+		if s[i] == ' ' || s[i] == '\t' {
+			needsQuotes = true
+			break
+		}
+	}
+
+	if runtime.GOOS != "windows" {
+		if needsQuotes {
+			return fmt.Sprintf(`"%s"`, s)
+		}
+		return s
+	}
+
+	var b []byte
+	slashes := 0
+	for i := 0; i < len(s); i++ {
+		if s[i] == '\\' {
+			slashes++
+			continue
+		}
+
+		if slashes == 1 {
+			b = append(b, '\\')
+		}
+		for ; slashes > 0; slashes-- {
+			b = append(b, '\\')
+		}
+		b = append(b, s[i])
```

Comment:
I looked at the PR on the weekend and I found this puzzling enough to leave until this morning.  I had to read through this a few times to understand it (high cyclomatic complexity).

WDYT of this simplification:
```suggestion
		switch s[i] {
		case '\\':
			slashes++

		default:
			// ensure a single slash is escaped
			if slashes == 1 {
				b = append(b, '\\')
			}
			slashes = 0
		}
		b = append(b, s[i])
```